openchs-server-api/src/main/java/org/openchs/service/ProgramEncounterService.java

```diff
@@ -44,4 +72,106 @@ public ProgramEncountersContract constructProgramEncounters(ProgramEncounter pro
         return  programEncountersContract;
     }
 
+    public ProgramEnrolment getAllEncountersForEnrolment(String uuid){
+        ProgramEnrolment programEnrolment = programEnrolmentRepository.findByUuid(uuid);
+        return programEnrolment;
+    }
+
+    public List<ProgramEncounter> constructVisitSchedules(ProgramEnrolment programEnrolment,String encounterTypeName){
+        return programEnrolment.getProgramEncounters().stream().filter(programEncounter -> programEncounter.getEncounterType().getOperationalEncounterTypeName().equals(encounterTypeName)).collect(Collectors.toList());
+    }
+
+    public void saveVisitSchedules(String uuid,List<VisitSchedule> visitSchedules){
+        for( VisitSchedule visitSchedule : visitSchedules){
+            try {
+                processVisitSchedule(uuid, visitSchedule);
+            }catch (Exception e){
+                RuleFailureLog ruleFailureLog = new RuleFailureLog();
```

Comment:
It will be nice if we create a static method like createInstance on RuleFailureLog which takes params. It will become a single line code here then.